py/dynruntime.h

```diff
@@ -52,22 +52,35 @@
 /******************************************************************************/
 // Memory allocation
 
+#define m_malloc_fail(num_bytes)        (m_malloc_fail_dyn((num_bytes)))
```

Comment:
This is kind of unrelated to this fix, but do these have to be macros here? It seems like the inline function could be named `m_malloc_fail`? Or is this necessary to "override" the default functions with these names?

(If there is a reason, suggest putting it in a comment.)

Reply:
The macro is necessary, because the existing `m_malloc_fail()` is declared as non-static, and so it'll clash here if declared `static inline`.

This pattern (using `xxx_dyn` functions with wrapper macros) is used a lot in this file, so any comment would need to apply to the whole file.

Reply:
Yes, a comment at the top of this file that explains what it's for and why everything follows this pattern would be really useful :)

Reply:
I've now added a comment, and slightly improved the new macros (a zero-byte allocation should not raise an exception even though it returns NULL).

Reply:
Thanks!